pages/index.js

```diff
@@ -1,30 +1,79 @@
-import React from "react";
+import React, { useState } from 'react';
+import styled from 'styled-components';
 
-import TodoInput from "../components/todo-input";
-import TodoList from "../components/todo-list";
-
-import styled from "styled-components";
+import TodoInput from '../components/todo-input';
+import TodoList from '../components/todo-list';
 
 export default function Home() {
-  return (
-    <Wrapper>
-      <Contents>
-        <TodoInput />
-        <TodoList />
-      </Contents>
-    </Wrapper>
-  );
+	const [date, setDate] = useState();
+	const [todo, setTodo] = useState();
+	const [todos, setTodos] = useState([]);
+	const [id, setID] = useState(0);
+
+	const changeTodo = (e) => {
+		setTodo(e.target.value);
+	};
+
+	const changeDate = (e) => {
+		setDate(e.target.value);
+	};
+
+	const removeTodo = (id) => {
+		const newTodos = [...todos];
+		setTodos(newTodos.filter((info) => info.id !== id));
+	};
+	const submitTodo = (e) => {
+		e.preventDefault();
+		if (todo === '' || date === '') {
+			alert('모든 항목을 입력해주세요!');
+			return false;
+		} else if (date.length != 8) {
+			alert('날짜를 올바른 형식으로 입력해주세요!');
+			return false;
+		} else alert('입력되었습니다!');
+		setTodos(todos.concat({ id: id, date: date, todo: todo }));
+		setTodo('');
+		setDate('');
+		setID(id + 1);
+	};
+
+	return (
+		<Wrapper>
+			<Subject>리액트-투두</Subject>
+			<Contents>
+				<TodoInput
+					date={date}
+					todo={todo}
+					onChangeDate={changeDate}
+					onChangeTodo={changeTodo}
+					onSubmit={submitTodo}
+				/>
+				<Space />
+				<TodoList data={todos} onRemove={removeTodo} />
+			</Contents>
+		</Wrapper>
+	);
 }
+const Space = styled.div`
+	flex: 0.3;
+`;
```

Comment:
```suggestion
const Space = styled.div`
	flex: 1;
`;
```
flex 값을 1로 하나 0.3으로 하나 차이는 없겠지만, 보통 이런 경우 1을 많이 씁니다! [참고](https://heropy.blog/2018/11/24/css-flexible-box/#flex)